### PR TITLE
Add learning rate scheduler and early stopping to FineTune LLMs

### PR DESCRIPTION
**Pull request by lucas-aixplain**
Task: M-5542693469

### CODE REVIEW

File: aixplain/modules/finetune/hyperparameters.py [hunk 1]
```diff
@@ -2,16 +2,30 @@
...
-    warmup_steps: int = 500
     generation_max_length: int = 225
     tokenizer_batch_size: int = 256
     gradient_checkpointing: bool = False
     gradient_accumulation_steps: int = 1
     max_seq_length: int = 4096
+    warmup_ratio: float = 0.0
+    warmup_steps: int = 0
+    early_stopping_patience: int = 1
+    lr_scheduler_type: SchedulerType = SchedulerType.LINEAR
```
Comment (thiago-aixplain) [inline aixplain/modules/finetune/hyperparameters.py:31]: Does @dataclass_json converts the enumerator to string automatically?

Comment (lucas-aixplain) [reply aixplain/modules/finetune/hyperparameters.py:31]: yes, by running the following line:
```python 
finetune = FinetuneFactory.create(str(uuid.uuid4()), dataset_list, model, hyperparameters=Hyperparameters(lr_scheduler_type=SchedulerType.CONSTANT))
```
We get the following request body:
```json
{
  "datasets": [
    {
      "datasetId": "65293795abed4d551b86baac",
      "trainPercentage": 100,
      "devPercentage": 0
    }
  ],
  "sourceModelId": "64e615671567f848804985e1",
  "parameters": {
    "hyperparameters": {
      "epochs": 4,
      "train_batch_size": 4,
      "eval_batch_size": 4,
      "learning_rate": 2e-05,
      "generation_max_length": 225,
      "tokenizer_batch_size": 256,
      "gradient_checkpointing": false,
      "gradient_accumulation_steps": 1,
      "max_seq_length": 4096,
      "warmup_ratio": 0.0,
      "warmup_steps": 0,
      "early_stopping_patience": 1,
      "lr_scheduler_type": "constant"
    }
  }
}
```

Comment (thiago-aixplain) [reply aixplain/modules/finetune/hyperparameters.py:31]: should you include these parameters in the functional test?

**Review by thiago-aixplain — APPROVED** [review]
